VideoLocker/src/main/java/org/edx/mobile/model/course/CourseComponent.java

```diff
@@ -0,0 +1,250 @@
+        this.name = name;
+    }
+
+    @Override
+    public boolean isGraded() {
+        return graded;
+    }
+
+    @Override
+    public void setGraded(boolean graded) {
+        this.graded = graded;
+    }
+
+    @Override
+    public boolean isGradedSubDAG() {
+        return gradedSubDAG;
+    }
+
+    @Override
+    public void setGradedSubDAG(boolean gradedSubDAG) {
+        this.gradedSubDAG = gradedSubDAG;
+    }
+
+    @Override
+    public String getBlockUrl() {
+        return blockUrl;
+    }
+
+    @Override
+    public void setBlockUrl(String url) {
+        this.blockUrl = url;
+    }
+
+    @Override
+    public String getWebUrl() {
+        return webUrl;
+    }
+
+    @Override
+    public void setWebUrl(String webUrl) {
+        this.webUrl = webUrl;
+    }
+
+    @Override
+    public BlockCount getBlockCount() {
+        return blockCount;
+    }
+
+    @Override
+    public void setBlockCount(BlockCount count) {
+        this.blockCount = blockCount;
+    }
+
+    @Override
+    public CourseComponent getParent() {
+        return parent;
+    }
+
+    @Override
+    public List<IBlock> getChildren() {
+        return (List) children;
+    }
+
+    @Override
+    public CourseComponent getRoot(){
+        return root;
+    }
+
+
+    public boolean isMobileSupported() {
+        return mobileSupported;
+    }
+
+    public void setMobileSupported(boolean mobileSupported) {
+        this.mobileSupported = mobileSupported;
+    }
+
+    public boolean isContainer(){
+        return children != null && children.size() > 0;
+    }
+
+    public List<CourseComponent> getChildContainers(){
+        List<CourseComponent> childContainers = new ArrayList<>();
+        if ( children != null ){
+            for(CourseComponent c : children){
+                if ( c.isContainer() )
+                    childContainers.add(c);
+            }
+        }
+        return childContainers;
+    }
+
+    public List<CourseComponent> getChildLeafs(){
+        List<CourseComponent> childLeafs = new ArrayList<>();
+        if ( children != null ){
+            for(CourseComponent c : children){
+                if ( !c.isContainer() )
+                    childLeafs.add(c);
+            }
+        }
+        return childLeafs;
+    }
+
+    public List<HasDownloadEntry> getVideos(){
+        List<HasDownloadEntry> videos = new ArrayList<>();
+        for(CourseComponent c : children){
+            videos.addAll( c.getVideos() );
+        }
+        return videos;
+    }
+
+    /**
+     * used for navigation.
+     * @return <code>true</code> if it is the last child of direct parent. or it does not has direct parent
+     *         <code>false</code> if it is not
+     */
+    public boolean isLastChild(){
+        if ( parent == null )
+            return true;
+        List<IBlock> sibling = parent.getChildren();
+        if ( sibling == null ) {
+            return false;  //it wont happen. TODO - should we log here?
```

Comment:
Maybe it would happen if the course gets updated? Is there something we'd need to do to handle that case?